webapp/src/components/jira_issue_selector/jira_issue_selector.jsx

```diff
@@ -55,9 +55,14 @@ export default class JiraIssueSelector extends Component {
     };
 
     searchIssues = (text) => {
-        const textEncoded = encodeURIComponent(text.trim().replace(/"/g, '\\"'));
-        const textSearchTerm = (textEncoded.length > 0) ? 'text ~ "' + textEncoded + '*"' : '';
-        const finalQuery = textSearchTerm + ' ' + searchDefaults;
+        const trimmed = text.trim();
+        if (trimmed === '') {
+            return {};
+        }
+        const textEncoded = encodeURIComponent(trimmed.replace(/"/g, '\\"'));
+        const textSearchTerm = 'text ~ "' + textEncoded + '*"';
```

Comment:
`2/5` propose `s/textKeyTerm/textIssueKeyTerm/` so the reading constructed `finalQuery` variable is more obvious.

Reply:
https://github.com/mattermost/mattermost-plugin-jira/pull/325/files#diff-7a8afa6c78a7c8d826b0b0a89d826925R65

Reply:
The better fix IMO is to move all this to the server, and re-define the API as "IssueAutocomplete", but I agree, will make this more intention-revealing.

Reply:
:+1: on moving to server